mm2src/coins/nft.rs

```diff
@@ -1574,3 +1702,56 @@ async fn build_and_send_request(
     let response = send_request_to_uri(uri, payload.as_deref()).await?;
     Ok(response)
 }
+
+pub async fn find_unique_nft_account_ids(
+    ctx: &MmArc,
+    chains: Vec<Chain>,
+) -> Result<HashMap<String, Vec<Chain>>, String> {
+    let cctx = try_s!(CoinsContext::from_ctx(ctx));
+    let coins = cctx.coins.lock().await;
+    let coins = coins.values().collect::<Vec<_>>();
+
+    let mut active_id_chains = HashMap::new();
+    for coin in coins.iter() {
+        if coin.is_available() {
+            // Use default if no db_id
+            let db_id = coin
+                .inner
+                .account_db_id()
+                .await
+                .unwrap_or_else(|| ctx.rmd160.to_string());
+            let entry = active_id_chains.entry(db_id).or_insert_with(Vec::new);
+            if let Ok(chain) = Chain::from_ticker(coin.inner.ticker()) {
+                if chains.contains(&chain) {
+                    entry.push(chain);
+                }
+            }
+        }
+    }
+
+    Ok(active_id_chains)
+}
```

Comment:
It could be simplified 
```rust
async fn find_unique_nft_account_ids(
    ctx: &MmArc,
    chains: Vec<Chain>,
) -> Result<HashMap<String, Vec<Chain>>, String> {
    let cctx = try_s!(CoinsContext::from_ctx(ctx));
    let coins = cctx.coins.lock().await;

    let mut active_id_chains: HashMap<String, Vec<Chain>> = HashMap::new();
    for coin in coins.values() {
        if coin.is_available() {
            // Use default if no db_id
            let db_id = coin
                .inner
                .account_db_id()
                .await
                .unwrap_or_else(|| ctx.rmd160.to_string());
            if let Ok(chain) = Chain::from_ticker(coin.inner.ticker()) {
                if chains.contains(&chain) {
                    active_id_chains.entry(db_id).or_default().push(chain);
                }
            }
        }
    }

    Ok(active_id_chains)
}
```

You can avoid collecting values into a Vec, and directly iterate over `coins.values()`.
instead of creating entry object, you can call `.entry(db_id).or_default()`.
Also seems like there is no need in `pub`